src/main/java/com/blackducksoftware/integration/hub/alert/web/SSLAuthenticationHandler.java

```diff
@@ -36,6 +36,10 @@
 
     @Override
     protected void configure(final HttpSecurity http) throws Exception {
-        http.requiresChannel().anyRequest().requiresSecure().and().csrf().disable().authorizeRequests().anyRequest().permitAll().and().logout().logoutSuccessUrl("/");
+        final String[] allowedPaths = { "/", "/built/bundle.js", "index.html", "/api/configuration/global", "/api/login", "/api/logout" };
+        http.requiresChannel().anyRequest().requiresSecure().and().csrf().disable().authorizeRequests().antMatchers(allowedPaths).permitAll().and()
```

Comment:
We can do permitAll() and hasRole("ADMIN")?

Reply:
yes the 'and' is essentially appending rules where the security context should be applied.  The first rule is that we requires SSL for any request.  Then for specific paths in the web app we permit all requests.  Then we add the rule to require the role of admin for any other request.  The permit all is applying to the paths specified in the antmatchers argument.

Reply:
That makes sense to me.